ipykernel/kernelbase.py

```diff
@@ -698,7 +698,7 @@ def raw_input(self, prompt=''):
             raise StdinNotImplementedError(
                 "raw_input was called, but this frontend does not support input requests."
             )
-        return self._input_request(str(prompt),
```

Comment:
@steff456, please use here the function called `cast_unicode_py2` from the module `ipython_genutils.py3compat`. That should solve the problems with the tests.